docs/projects/trenchboot-aem.md

```diff
@@ -0,0 +1,301 @@
+
+    - Test the solution on AMD hardware with TPM 2.0 and TPM 1.2 with legacy boot
+      mode
+
+4. Phase 4 - Xen UEFI boot mode with DRTM:
+
+    - TrenchBoot support for UEFI boot mode for AMD in GRUB
+
+        While TrenchBoot DRTM was extensively tested on Intel hardware with UEFI
+        firmware and Linux, it was not on AMD platforms. This task ensures that
+        DRTM works with UEFI boot mode on AMD processors in GRUB2 and Linux without
+        UEFI Boot Services.
+
+    - TrenchBoot support for UEFI boot mode in Xen
+
+        When UEFI boot mode with TrenchBoot is working with GRUB2 and Linux, all
+        that is missing to fully support AMD and Intel hardware with Qubes OS AEM
+        is the Xen support to boot in UEFI mode without Boot Services. This
+        requires a significant amount of work to make sure that all information
+        that Xen would obtain from UEFI Boot Services would be still available. The
+        information has to be passed by GRUB2 to Xen via Multiboot2 protocol:
+
+         + EFI memory map
+         + Framebuffer information
+         + PCI devices information with their option ROMs
+
+        Additionally "EFI boot services not terminated" Multiboot2 tag must not be
+        passed to Xen by GRUB2 on DRTM launch when GRUB2 will terminate Boot
+        Services. Xen should detect such situation and act according to the state
+        of Boot Services. Xen will be implemented to:
+
+         + parse the EFI memory map, framebuffer information and PCI devices
+           information passed by GRUB2
+         + do not expose the "EFI boot services" Multiboot2 tag indicating that
+           Xen can be executed without UEFI Boot Services presence
+         + allocate the memory space for the trampoline used to launch other
+           processors or use the allocation done by GRUB2 if necessary
+         + do not go error path when Boot Services are not present and skip all
+           calls to UEFI Boot Services by using the information provided by GRUB2
+
+    - Test the solution on AMD and Intel hardware with TPM 2.0 and TPM 1.2 with
+      legacy and UEFI boot mode
+
+## Projects or organizations relevant to this project before?
+
+1. [Qubes OS](https://www.qubes-os.org/)
+2. [Xen Hypervisor](https://xenproject.org/)
+3. [GNU GRUB](https://www.gnu.org/software/grub/)
+4. [TrenchBoot](https://trenchboot.org/)
+5. [Invisible Things Lab](https://invisiblethingslab.com/)
+6. [Apertus Solutions](https://apertussolutions.com/)
+7. [Oracle](https://www.oracle.com/)
+8. [3mdeb](https://3mdeb.com/)
+
+## The ecosystem of the project
+
+3mdeb has a good relationship with the maintainers of relevant projects which
+will participate in review of the work:
+
+- Marek Marczykowski-Górecki (Invisible Things Lab CTO) - Qubes OS maintainer
+- Andrew Cooper (Citrix) - Xen Hypervisor Maintainer
+- Daniel Kiper (Oracle) - GRUB2 Maintainer
+- Daniel Smith (Apertus Solutions) - TrenchBoot founder and maintainer
```

Comment:
Why not say how further review will be continued? That people can use Giscus or that they can contribute directly to this repository.

Reply:
Done